src/amperity/gocd/secret/vault/plugin.clj

```diff
@@ -158,23 +157,26 @@
         (some #(% field-value) (:validate-fns field-model))))))
 
 
+(defn- client-from-inputs
+  [inputs]
+  (component/start
```

Comment:
i think its ok _not_ starting the component here, since the client will be so short lived. This sets up lease expiration watchers which are irrelevant when the client just gets torn down right away.

Reply:
I'll second this, there's no need to start the client here and if you forget to stop it somewhere you'll be leaking threads and client references. Seems like this could also be inlined into `authenticate-client-from-inputs!` below?